tests/benchmark/test_worst_blocks.py

```diff
@@ -156,3 +157,69 @@ def test_block_full_of_ether_transfers(
         blocks=[Block(txs=txs)],
         exclude_full_post_state_in_output=True,
     )
+
+
+@pytest.fixture
+def total_cost_floor_per_token():
+    """Total cost floor per token."""
+    return 10
+
+
+@pytest.mark.valid_from("Prague")
+@pytest.mark.parametrize("zero_byte", [True, False])
+def test_block_full_of_empty_payload(
+    state_test: StateTestFiller,
+    pre: Alloc,
+    zero_byte: bool,
+    intrinsic_cost: int,
+    total_cost_floor_per_token: int,
+):
+    """Test a block with empty payload."""
+    attack_gas_limit = Environment().gas_limit
+
+    # Gas cost calculation based on EIP-7683: (https://eips.ethereum.org/EIPS/eip-7683)
+    #
+    #   tx.gasUsed = 21000 + max(
+    #       STANDARD_TOKEN_COST * tokens_in_calldata
+    #       + execution_gas_used
+    #       + isContractCreation * (32000 + INITCODE_WORD_COST * words(calldata)),
+    #       TOTAL_COST_FLOOR_PER_TOKEN * tokens_in_calldata)
+    #
+    # Simplified in this test case:
+    # - No execution gas used (no opcodes are executed)
+    # - Not a contract creation (no initcode)
+    #
+    # Therefore:
+    #   max_token_cost = max(STANDARD_TOKEN_COST, TOTAL_COST_FLOOR_PER_TOKEN)
+    #   tx.gasUsed = 21000 + tokens_in_calldata * max_token_cost
+    #
+    # Since max(STANDARD_TOKEN_COST, TOTAL_COST_FLOOR_PER_TOKEN) = 10:
+    #   tx.gasUsed = 21000 + tokens_in_calldata * 10
+    #
+    # Token accounting:
+    #   tokens_in_calldata = zero_bytes + 4 * non_zero_bytes
+    #
+    # So we calculate how many bytes we can fit into calldata based on available gas.
+
+    gas_available = attack_gas_limit - intrinsic_cost
+
+    # Calculate the token_in_calldata
+    gas_available //= total_cost_floor_per_token
+
+    # Calculate the number of bytes that can be stored in the calldata
+    num_of_bytes = gas_available // 4 if zero_byte else attack_gas_limit
```

Comment:
```suggestion
    # Calculate the token_in_calldata
    max_tokens_in_calldata = gas_available // total_cost_floor_per_token

    # Calculate the number of bytes that can be stored in the calldata
    num_of_bytes = max_tokens_in_calldata if zero_byte else max_tokens_in_calldata // 4
```
Suggest two changes:
- Use a `max_tokens_in_calldata` instead of `gas_limit` when `//4` since after doing that division, "gas limit" semantics seems to be lost since we aren't talking about gas anymore but tokens.
- I think your previous formula (i.e., `num_of_bytes = gas_available // 4 if zero_byte else attack_gas_limit`) was maybe incorrect?. We should `//4` if `not zero_byte`, and also under no case we should us `attack_gas_limit` as the number of bytes?